### PR TITLE
Change basemap style without reloading the page. Closes #300

### LINKED ISSUE #300: Changing basemap clears other layers
As tested just now...

### PR DESCRIPTION
**Pull request by dabreegster**
This adapts the work @stuartlynn recently added to `svelte-maplibre` to let us change MapLibre basemap without reloading the entire page. This is a _much_ better UX to not lose current page state, and made even nicer with the new basemaps added or coming soon.

Demos: https://acteng.github.io/atip/swap_basemap/browse.html

https://github.com/acteng/atip/assets/1664407/14dafc4a-d9f0-4968-b157-831d7ee52470

https://github.com/acteng/atip/assets/1664407/5b401dca-b46b-4ba6-9140-bb2783a84245

CC @mem48 in case this helps with NPT

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "govuk-frontend": "^4.6.0",
     "humanize-string": "^3.0.0",
     "js-cookie": "^3.0.5",
-    "maplibre-gl": "^3.1.0",
+    "maplibre-gl": "^3.3.1",
```
Comment (dabreegster) [inline package.json:53]: Not strictly part of this change and can revert / do separately if useful, but while trying to figure out `setStyle` weirdness, I upgraded. From the changelog, nothing looks relevant for us, and can't hurt to be modern

**Comment by robinlovelace-ate** [comment]
Looks good to me, but other layers still 'flicker', right?

**Comment by stuartlynn** [comment]
> Looks good to me, but other layers still 'flicker', right?

Yeah not sure if there is a good way around this. The only way I think to prevent that is to not use maplibres setStyle method and instead fetch the style ourselves, remove the previous style layers and sources and then add the new styles while leaving the user defined ones intact. Though I am not even 100% that would fix the flicker.

Might be woth investigating though

File: src/lib/common/MapLibreMap.svelte [hunk 2]
```diff
@@ -46,11 +58,57 @@
...
+        (sourceId) => !lastStyleSourceIds!.includes(sourceId)
+      );
+      sourcesToReAddAfterStyleChange = {};
+      for (const id of nonStyleSourceIds) {
+        sourcesToReAddAfterStyleChange[id] = oldMapStyle.sources[id];
+      }
+    }
+    map.setStyle(await getStyleSpecification(newStyle), { diff: false });
+
+    // Update the URL query parameter
```
[the remark this answers is not in the document]
Comment (stuartlynn) [reply src/lib/common/MapLibreMap.svelte:106]: I think that the changes have been merged and released upstream so we should be able to remove this and just pull in the new version of svelte-maplibre

[the remark this answers is not in the document]
Comment (stuartlynn) [reply src/lib/common/MapLibreMap.svelte:106]: Ah got it!

**Review by Pete-Y-CS — APPROVED** [review]

**Pull request merged**